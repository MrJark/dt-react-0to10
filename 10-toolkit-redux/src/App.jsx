import { useDispatch, useSelector } from 'react-redux';


import reactLogo from './assets/react.svg';
import viteLogo from '/vite.svg';
import './App.css';
import { increment, decrement, incrementBy } from './store/slices/counter';

function App() {

  const { counter } = useSelector( state => state.counter);
  const dispatch = useDispatch();
  const numByIncrement = 2;
  
  return (
    // tarea: hacer el decrement function y el increment by 2 donde el 2 tiene que ser un argumento que se pueda modificar dependiendo del incremento. Que no sea estático
    // incremento ✅ y el incrementBy ❌ me salía el value como null y no sabía como cambiarlo
    // lo había hecho bien solo que al copiar del ejemplo, no había cambiado el value por el counter y claro, no existía
    <>
      <div>
        <a href="https://vitejs.dev" target="_blank">
          <img src={viteLogo} className="logo" alt="Vite logo" />
        </a>
        <a href="https://react.dev" target="_blank">
          <img src={reactLogo} className="logo react" alt="React logo" />
        </a>
      </div>
      <h2>count is {counter}</h2>
      <div className="card">
        <button type='button' style={{ margin: 4}} onClick={() => {dispatch( increment() )}}>
          Increment
        </button>
        <button type='button' style={{ margin: 4}} onClick={() => {dispatch( decrement() )}}>
          Decrement
        </button>
        <button type='button' style={{ margin: 4}} onClick={() => {dispatch( incrementBy(numByIncrement) )}}>
          Increment by {numByIncrement}
        </button>
      </div>
    </>
  )
}

export default App
