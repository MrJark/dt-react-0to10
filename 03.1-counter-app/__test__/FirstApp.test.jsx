import {render} from '@testing-library/react';
import App from '../src/FirstApp';


describe('pruebas en <App /> de FirstApp', () => {

    test('debe hacer match con el snapchot', () => {

        const title = 'Hola, soy Vegeta';
        const {container} = render( <App title={title}/> ); // render además exporta un objeto que tiene ciertas propiedades, como en este caso el container

        expect(container).toMatchSnapshot(); // este método crea un snapchot del modelo que estás testeando para que se algo se pierde por el camino, puedas saber como era antes
    });

    test('debe mostrar el título como un h1', () => {

        const title = 'Hola soy Vegeta';
        const { container, getByText } = render( <App title={title}/> );

        expect( getByText(title) ).toBeTruthy(); // coge el title y, como texto que es, espera que exita con en toBeTruthy ( no cumple esto solo la prueba)

        const h1 = container.querySelector('h1'); // container actua como document en js
        console.log(h1.innerHTML);
        // expect(h1.innerHTML).toBe(title); // el problema de este es que si hay espacios en algún lugar y no es exactamente igual, dará errer. la solución -> .toContain
        expect(h1.innerHTML).toContain(title);
    })

})