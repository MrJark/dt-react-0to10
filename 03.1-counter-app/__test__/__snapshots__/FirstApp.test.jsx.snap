// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <App /> de FirstApp debe hacer match con el snapchot 1`] = `
<div>
  <h1>
    Hola, soy Vegeta
  </h1>
  <h1>
    No hay subtitulo
  </h1>
  <h2>
    Chema Ferrandez
  </h2>
</div>
`;
